File: api/src/main/java/org/pmiops/workbench/audit/adapters/ProfileAuditAdapterService.java
package org.pmiops.workbench.audit.adapters;

import org.pmiops.workbench.model.Profile;

public interface ProfileAuditAdapterService {
  void fireCreateAction(Profile createdProfile);

  void fireUpdateAction(Profile previousProfile, Profile updatedProfile);

  void fireDeleteAction(long userId, String userEmail);
}

File: api/src/main/java/org/pmiops/workbench/audit/adapters/ProfileAuditAdapterService.kt
package org.pmiops.workbench.audit.adapters

import org.pmiops.workbench.model.Profile

interface ProfileAuditAdapterService {
    fun fireCreateAction(createdProfile: Profile)

    fun fireUpdateAction(previousProfile: Profile, updatedProfile: Profile)

    fun fireDeleteAction(userId: Long, userEmail: String)
}
